### PR TITLE
Normalize identifiers to `-` instead of `_`

### PR DESCRIPTION
**Pull request by ogoffart**
As a result
 - The error messages will now show the error with `-` instead of `_`
 - The LSP will auto-complete with -
 - The interpreter's list of properties will list the property with '-'
   (but we made the change so that set_property, get_property, and so on
   work also if passed a '-')

### CODE REVIEW

**Comment by ogoffart** [comment]
cc #382

File: docs/langref.md [hunk 1]
```diff
@@ -118,7 +118,7 @@ C-style comments are supported:
 
 Identifiers can be composed of letter (`a-zA-Z`), of numbers (`0-9`), or of the underscore (`_`) or the dash (`-`).
 They cannot start with a number or a dash (but they can start with underscore)
-The dashes are normalized to underscore. Which means that these two identifiers are the same: `foo-bar` and `foo_bar`.
+The underscore are normalized to dashes. Which means that these two identifiers are the same: `foo_bar` and `foo-bar`.
```
Comment (tronical) [inline docs/langref.md:121]:
```suggestion
The underscores are normalized to dashes. Which means that these two identifiers are the same: `foo_bar` and `foo-bar`.
```

File: docs/langref.md [hunk 2]
```diff
@@ -787,11 +787,11 @@ of assigning a different name at import time:
 
 ```60,ignore
 import { Button } from "./button.60";
-import { Button as CoolButton } from "../other_theme/button.60";
+import { Button as CoolButton } from "../other-theme/button.60";
```
Comment (tronical) [inline docs/langref.md:790]:
```suggestion
import { Button as CoolButton } from "../other_theme/button.60";
```

I think dashes in directory names are not so common.

File: docs/langref.md [hunk 3]
```diff
@@ -787,11 +787,11 @@ of assigning a different name at import time:
...
 
 App := Rectangle {
     // ...
-    CoolButton {} // from cool_button.60
+    CoolButton {} // from cool-button.60
```
Comment (tronical) [inline docs/langref.md:794]:
```suggestion
    CoolButton {} // from other_theme/button.60
```

File: sixtyfps_compiler/builtins.60 [hunk 4]
```diff
@@ -568,9 +568,9 @@ export NativeTab := _ {
...
 export global NativeStyleMetrics := {
-    property <length> layout_spacing;
-    property <length> layout_padding;
-    property <length> text_cursor_width;
+    property <length> layout-spacing;
+    property <length> layout-padding;
+    property <length> text-cursor-width;
     //-is_non_item_type
     //-is_internal
 }
```
Comment (tronical) [inline sixtyfps_compiler/builtins.60:576]: Just a thought, not relevant for this change: When loading `builtins.60` we could try to debug_assert that all property names/callbacks are free of underscores.

File: tools/syntax_updater/main.rs [hunk 5]
```diff
@@ -242,5 +242,12 @@ fn fold_token(
             return write!(file, "{}ms", node.text());
         }
     }*/
+    /* Example: replace _ by - in identifiers
+    if node.kind() == SyntaxKind::Identifier
+        && node.text().contains('_')
+        && !node.text().starts_with("_")
+    {
+        return file.write_all(node.text().replace('_', "-").as_bytes())
+    }*/
```
Comment (tronical) [inline tools/syntax_updater/main.rs:251]: Ahh, cool! This answers a question I was about to ask ;-)

**Review by tronical — APPROVED** [review]

File: sixtyfps_compiler/builtins.60 (hunk 4 above)
Comment (ogoffart) [reply sixtyfps_compiler/builtins.60:576]: we could...  but that's just style, this file is normalized anyway.

Another thought is that we could assert that in the documentation or in the example.

File: examples/iot-dashboard/iot-dashboard.60 [hunk 6]
```diff
@@ -409,19 +409,19 @@ BoxWithButtons := Box {
...
     iconFile: @image-url("images/indoor-temperature.png");
     value: (temperature < 0 ? "" : "+") + temperature;
     isBright: true;
     changed(delta) => { temperature += delta; }
 }
 export Humidity := BoxWithButtons {
-    property <int> humidity_percent : 30;
-    title_: "Humidity";
+    property <int> humidity-percent : 30;
+    title-: "Humidity";
```
Comment (ogoffart) [inline examples/iot-dashboard/iot-dashboard.60:420]: this looks a bit strange

Comment (tronical) [reply examples/iot-dashboard/iot-dashboard.60:420]: Ahh, because `Box` also has a `title` property and it's an error to override it :-(

**Review by tronical — APPROVED** [review]

**Pull request merged**